internal/controllers/admission_test.go

```diff
@@ -19,159 +19,178 @@ func operator(spec operatorsv1alpha1.OperatorSpec) *operatorsv1alpha1.Operator {
 	}
 }
 
-var _ = Describe("Operator Spec Validations", func() {
-	var (
-		ctx    context.Context
-		cancel context.CancelFunc
-	)
-	BeforeEach(func() {
-		ctx, cancel = context.WithCancel(context.Background())
-	})
-	AfterEach(func() {
-		cancel()
-	})
-	It("should fail if the spec is empty", func() {
-		err := cl.Create(ctx, operator(operatorsv1alpha1.OperatorSpec{}))
-		Expect(err).To(HaveOccurred())
-		Expect(err.Error()).To(ContainSubstring("spec.packageName in body should match '^[a-z0-9]+(-[a-z0-9]+)*$'"))
-	})
-	It("should fail if package name length is greater than 48 characters", func() {
-		err := cl.Create(ctx, operator(operatorsv1alpha1.OperatorSpec{
-			PackageName: "this-is-a-really-long-package-name-that-is-greater-than-48-characters",
-		}))
-		Expect(err).To(HaveOccurred())
-		Expect(err.Error()).To(ContainSubstring("Too long: may not be longer than 48"))
-	})
-	It("should fail if version is valid semver but length is greater than 64 characters", func() {
+func TestOperatorSpecIsEmpty(t *testing.T) {
```

Comment:
Looks like you would benefit from a `TestOperatorCreate` with table-driven sub-tests (running in parallel?)

Reply:
Possibly? This was more of a direct translation. Table-based tests have their pros and cons, and I didn't want to modify the set of tests while also changing the testing infrastructure.

Basically, I wanted to translate the tests to get rid of Ginkgo, without messing with the tests themselves too much, because it's just too easy to accidentally delete or subtly break a test. Optimizations like that can be done later, iteratively.

Reply:
Table-driven tests are the gold standard and have deep integrations with the Go standard `testing` library. The tests here are obvious candidates for that approach - I think a) removing Ginkgo in and of itself is not really the goal, but getting this into a state that's more aligned with the Go ecosystem is, and b) as such, choices _not_ to do the common thing (here, table tests) need justification.

Reply:
My point was more toward an incremental rather than all-in-one approach. I'm fine with table-based tests, _if they're appropriate_.